src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1831,3 +1831,32 @@ def set_attention_slice(self, slice_size: Optional[int]):
 
         for module in modules:
             module.set_attention_slice(slice_size)
+
+    def display_total_parameters(self):
+        r"""
+        Gives a breakdown of the number of parameters each `torch.nn.Module` component has. Also shows the sum of the
+        parameters each module-level component has.
+
+        Examples:
+
+        ```py
+        >>> from diffusers import DiffusionPipeline
+
+        >>> pipe = DiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-2-1")
+        >>> pipe.display_total_parameters()
+        ```
+        """
+
+        def format_dict(dict):
+            return "\n".join([f"- {prop}: {val} ({val / 1e6} Million)" for prop, val in dict.items()]) + "\n"
+
+        num_params_breakup = {
+            name: component.num_parameters()
+            for name, component in self.components.items()
+            if isinstance(component, torch.nn.Module)
+        }
+        total_params = sum(params for _, params in num_params_breakup.items())
+
+        print("\nBreakdown of the number of the parameters for each `torch.nn.Module` component of the pipeline:\n")
+        print(format_dict(num_params_breakup))
+        print(f"Total number of parameters the pipeline has: {total_params} ({total_params / 1e6} Million).\n")
```

Comment:
```suggestion
```

I think we can leave printing out the values to the user. IMO it'd be nicer to just return a dict of the params here

Reply:
In that case, the function just becomes a one-liner and I don't see any point in having it as a part of the class in the first place.